ruby/bps-stan/lib/bps/stan.rb

```diff
@@ -1,12 +1,55 @@
 require 'bps'
 require 'bps/publisher/stan'
+require 'bps/subscriber/stan'
 
 module BPS
   module Publisher
     register('stan') do |url, **opts|
-      cluster_id, client_id, url_opts = STAN.parse_url(url)
+      cluster_id, client_id, url_opts = ::BPS::STAN::Utils.parse_url(url)
       url_opts.update(opts)
-      STAN.new(cluster_id, client_id, **STAN.coercer.coerce(url_opts))
+      STAN.new(cluster_id, client_id, **::BPS::STAN::Utils.coercer.coerce(url_opts))
+    end
+  end
+
+  module Subscriber
+    register('stan') do |url, **opts|
+      cluster_id, client_id, url_opts = ::BPS::STAN::Utils.parse_url(url)
+      url_opts.update(opts)
+      STAN.new(cluster_id, client_id, **::BPS::STAN::Utils.coercer.coerce(url_opts))
+    end
+  end
+
+  module STAN
+    module Utils
```

Comment:
Not sure if this file is the best place to keep these common utils, but they're, well, really **common**